Yafc/Windows/WelcomeScreen.cs

```diff
@@ -198,6 +206,30 @@ protected override void BuildContents(ImGui gui) {
                 }
             }
         }
+        else if (definesUpdateState != DefinesUpdateState.None) {
+            gui.BuildText(LSs.LoadErrorNeedsNewDefines, TextBlockDisplayStyle.WrappedText);
+            using (gui.EnterRow()) {
+                if (gui.BuildButton(LSs.CopyUpdateDefinesCommand, definesUpdateState == DefinesUpdateState.WaitingForCopy ? SchemeColor.Primary : SchemeColor.Grey)) {
+                    SDL.SDL_SetClipboardText(DefinesUpdateCommand);
+                    definesUpdateState = DefinesUpdateState.WaitingForFile;
+                }
+                if (gui.BuildButton(LSs.MenuRunFactorio, definesUpdateState == DefinesUpdateState.WaitingForFile ? SchemeColor.Primary : SchemeColor.Grey)) {
+                    MainScreen.StartFactorio();
+                }
+                if (tryFileCopy()) {
+                    errorMessage = null;
+                    definesUpdateState = DefinesUpdateState.None;
+                    LoadProject();
+                }
+                ButtonEvent evt = gui.RemainingRow().BuildButton(LSs.BackButton, SchemeColor.Grey);
+                // "C:/<whatever>" looks wrong, but not replacing the \ risks trying to display unescaped '\n's.
+                _ = evt.WithTooltip(gui, LSs.LoadErrorNotYetUpdated.L(Path.Combine(Environment.CurrentDirectory, "Data").Replace('\\', '/')));
```

Comment:
It is not entirely clear what you are replacing, but with a regex you can filter on the (only) thing you (do not) want to replace? Then the Windows file path is not messed up...?

If this is not a big issue, this is fine as well, but you comment suggests you do not like it... :wink:

Reply:
I didn't consider replacing `\n` with `/n`, but I think I like that less, since that means we get a mix of path separators on Windows.

- No replacement: ... the folder C:\\...\yafc\Yafc\bin\Debug
et8.0\Data
- Replacing \n only: ... the folder C:\\...\yafc\Yafc\bin\Debug/net8.0\Data
- Replace all \\: ... the folder C:/.../yafc/Yafc/bin/Debug/net8.0/Data

What really needs to happen here is better handling of escaping, but I'm not sure how much I want to do that when Factorio interprets `\\n` as "draw a backslash, then move to a new line" instead of "draw a backslash, then draw an n".

Reply:
> Factorio interprets \\n as "draw a backslash

Really? :open_mouth: They should consider that as a bug IMO.

But I agree with you that the current implementation is the best (without fully/better handling of escaping)